spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
+            It will be shown at startup in a :class:`QMessageBox`.
         """
         valid = True
-        message = ''  # Note: Remember to use _('') to localize the string
+        message = ""  # Note: Remember to use _('') to localize the string
         return valid, message
 
-    def on_first_registration(self):
+    def on_first_registration(self) -> None:
         """
         Actions to be performed the first time the plugin is started.
 
         It can also be used to perform actions that are needed only the
         first time this is loaded after installation.
 
         This method is called after the main window is visible.
+
+        Returns
+        -------
+        None
         """
         pass
 
-    def before_mainwindow_visible(self):
+    def before_mainwindow_visible(self) -> None:
         """
-        Actions to be performed after setup but before the main window's has
-        been shown.
+        Actions to be performed after setup but before showing the main window.
+
+        Returns
+        -------
+        None
         """
         pass
 
-    def on_mainwindow_visible(self):
+    def on_mainwindow_visible(self) -> None:
         """
-        Actions to be performed after the main window's has been shown.
+        Actions to be performed after the main window has been shown.
+
+        Returns
+        -------
+        None
         """
         pass
 
-    def on_close(self, cancelable=False):
+    def on_close(self, cancelable=False) -> None:
         """
         Perform actions before the plugin is closed.
 
-        This method **must** only operate on local attributes and not other
-        plugins.
+        .. caution::
+
+            This method **must** only operate on local attributes and not
+            call other plugins.
+
+        Parameters
+        ----------
+        cancelable: bool, optional
+            ``True`` if the close operation can potentially canceled;
+            ``False`` by default.
+
+        Returns
+        -------
+        None
         """
-        if hasattr(self, 'unregister'):
-            warnings.warn('The unregister method was deprecated and it '
-                          'was replaced by `on_close`. Please see the '
-                          'Spyder 5.2.0 migration guide to get more '
-                          'information.')
+        pass
 
     def can_close(self) -> bool:
         """
-        Determine if a plugin can be closed.
+        Determine if the plugin can be closed.
 
         Returns
         -------
         close: bool
-            True if the plugin can be closed, False otherwise.
+            ``True`` if the plugin can be closed, ``False`` otherwise.
         """
         return True
 
-    def update_font(self):
+    def update_font(self) -> None:
         """
-        This must be reimplemented by plugins that need to adjust their fonts.
+        Reimplemented by plugins that need to adjust their fonts.
```

Comment:
```suggestion
        This mus be reimplemented by plugins that need to adjust their fonts.
```

It's hard to grasp what this docstring means with the change you made.

Reply:
Actually, I think I meant to go back and add a proper summary line to these after the initial pass, but neglected to do so.

Adding that as well as restoring this to how it was

```suggestion
        Modify the font used in the plugin's interface.
        
        This must be reimplemented by plugins that need to adjust their fonts.
```